src/ServerRequestFactory.php

```diff
@@ -99,6 +101,52 @@ public static function fromGlobals(
         );
     }
 
+    /**
+     * Create a request from a Swoole request
+     *
+     * @param swoole_http_request $request
+     * @return ServerRequest
+     * @throws RuntimeException if Swoole is not installed
+     */
+
+    public static function fromSwoole(swoole_http_request $request)
+    {
+        if (! extension_loaded('swoole')) {
```

Comment:
Doesn't look like a good idea: as commented by @moln, this should be a hard requirement, with an adapter package enforcing it

Reply:
@Ocramius, @weierophinney maybe [zend-expressive-swoole](https://github.com/zendframework/zend-expressive-swoole/) is a better fit for this PR?

Reply:
:+1: